events/src/camera.rs

```diff
@@ -66,8 +102,128 @@ pub struct Capability {
 }
 
 /// Type representing rational number.
-#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct Fraction {
     pub numer: i32,
     pub denom: i32,
 }
+
+/// Camera status.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum Status {
+    Online,
+    Offline,
+}
+
+/// Local camera interface.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum LocalCameraInterface {
+    Usb,
+    Csi,
+}
+
+pub trait CameraBase {
```

Comment:
Do we really need this trait or should these maybe just be methods on the enum? If we keep it, we should probably implement it for the enum too.

Reply:
> Do we really need this trait or should these maybe just be methods on the enum?

from experience with porting of API to `pipeline` crate, I can tell that it can make it very convenient to write generic code that deals uniformly with different kinds of cameras. While it may not be immediately useful, this is public API and we need to try and anticipate future uses/needs a bit.

>  If we keep it, we should probably implement it for the enum too.

Makes sense, yes.

Reply:
> from experience with porting of API to `pipeline` crate, I can tell that it can make it very convenient to write generic code that deals uniformly with different kinds of cameras. While it may not be immediately useful, this is public API and we need to try and anticipate future uses/needs a bit.

Sorry, what I wrote was a bit unclear. What I wanted to say is that in this case, code that wants to operate on different kinds of cameras could just use the `Camera` type (assuming it does get the methods that are currently on the trait).

Reply:
Please have a look at enum impl, there are methods `as_base` and `as_base_mut` that can be used to access these methods quite easily.

Regarding enum methods vs trait: I don't know really which is better. If I make a bunch of methods on enum then I have to match all variants in each method. Probably it can be simplified a bit with some macro kung fu.

Reply:
Here is a WIP [use case](https://github.com/lumeohq/lumeod/blob/1e1ce121f07a6b4d292658a3935c7b99244377cf/src/lumeod.rs#L222). I use `status()` to process cameras without matching the exact variant. And no need to implement trait for the enum at the cost of calling `as_base()` at user side.

Reply:
Implemented `CameraBase` for `Camera`, please have a look.